### PR TITLE
add a new property Name to MultipartItem class

### PR DESCRIPTION
**Pull request by javirszhang**
consider following suituation, my webapi get upload files from Request.Form.Files so that IFormFile.Name is very important.
```
[HttpPost("upload/{bucket}")]
public ApiResult<List<FileUploadResponseDTO>> FileUpload(string bucket)
{
    List<FileUploadResponseDTO> list = new List<FileUploadResponseDTO>();
    MediaFileUploadProvider mediaFile = new MediaFileUploadProvider(bucket);
    foreach (var file in Request.Form.Files)
    {
        var fileDTO = mediaFile.Execute(file.Name, file.FileName, file.OpenReadStream());
        if (fileDTO != null)
        {
            list.Add(fileDTO);
        }
    }
    return list;
}
```
IFormFile.Name determins file means, it can not use a fixed parameter name.

**What kind of change does this PR introduce?**
enhancement, let users can set Content-Disposition name  attribute when necessary.

**What is the current behavior?**
Always use the parameter name as the Content-Disposition name attribute.

**What is the new behavior?**
Add a new property Name to MultipartItem  class, so we can set it when necessary.

**What might this PR break?**
nothing

**Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/reactiveui/refit?pullRequest=973) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/reactiveui/refit?pullRequest=973) before we can accept your contribution.<br/><br/>:x: javirszhang [sign now](https://cla.dotnetfoundation.org/reactiveui/refit?pullRequest=973)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/reactiveui/refit?pullRequest=973) it.</sub>

File: Refit/MultipartItem.cs [hunk 1]
```diff
@@ -7,12 +7,13 @@ namespace Refit
 {
     public abstract class MultipartItem
     {
-        public MultipartItem(string fileName, string contentType)
+        public MultipartItem(string fileName, string contentType, string name)
```
Comment (clairernovotny) [inline Refit/MultipartItem.cs:10]: Can you please add another ctor so it's not a breaking change?

Comment (javirszhang) [reply Refit/MultipartItem.cs]: It's a better idea, I push a new commit for it. thank you.

**Comment by clairernovotny** [comment]
Included in #1002

**Pull request closed without merging**